service_management/azure/test/integration/vm/vm_create_test.rb

```diff
@@ -0,0 +1,300 @@
+# Copyright 2013 Microsoft Open Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#--------------------------------------------------------------------------
+require 'integration/test_helper'
+
+describe Azure::VirtualMachineManagementService do
+  subject { Azure::VirtualMachineManagementService.new }
+  let(:cloud_mgmt_service) { Azure::CloudServiceManagementService.new }
+  let(:vm_name) { 'testvm2321v' }
+  let(:linux_vm_image) { '03f55de797f546a1b29d1b8d66be687a__CoreCLR-x64-Beta5-Linux-PartsUnlimited-Demo-App-201504.29' }
+  let(:windows_vm_image) { '03f55de797f546a1b29d1b8d66be687a__Team-Foundation-Server-2013-Update4-WS2012R2' }
+  let(:geo_location) { 'Southeast Asia' }
+  let(:cloud_service) { 'testcloudservice2321v' }
+  let(:storage_account) { 'sa4375943275v' }
+  let(:username) { 'adminuser' }
+  let(:password) { 'Admin123' }
+  let(:private_key) { Fixtures['privatekey.key'] }
+  let(:params) do
+    {
+      vm_name: vm_name,
+      vm_user: username,
+      image: linux_vm_image,
+      password: password,
+      location: geo_location
+    }
+  end
+  let(:windows_params) do
+    {
+      vm_name: vm_name,
+      vm_user: username,
+      image: windows_vm_image,
+      password: password,
+      location: geo_location
+    }
+  end
+  let(:options) do
+    {
+      storage_account_name: storage_account,
+      cloud_service_name: cloud_service,
+      vm_size: 'Basic_A0'
+    }
+  end
+
+  before do
+    Azure::Loggerx.expects(:puts).returns(nil).at_least(0)
+  end
+
+  describe 'vm add role' do
+    before do
+      VCR.insert_cassette "vm/#{name}"
+      @vm_obj = subject.create_virtual_machine(params, options)
+    end
+
+    after do
+      subject.delete_virtual_machine(vm_name, cloud_service)
+      Azure::StorageManagementService.new.delete_storage_account(storage_account)
+      VCR.eject_cassette
+    end
+
+    it 'should add role to existing storage account and cloud service' do
+      params[:cloud_service_name] = options[:cloud_service_name]
+      options.delete(:cloud_service_name)
+      params.delete(:location)
+      params[:vm_name] = "add-#{vm_name}"
+      vm = subject.add_role(params, options)
+      vm.cloud_service_name.must_equal params[:cloud_service_name]
+      vm.vm_name.must_equal params[:vm_name]
+      vm.deployment_name.must_equal @vm_obj.deployment_name
+      vm.os_type.must_equal 'Linux'
+      subject.delete_virtual_machine(params[:vm_name], cloud_service)
```

Comment:
Actually in this case yes. When we add a role in the VM it creates two vm as resources. Line 81 removes the vm created with role and line 66 removed original vm.